addons/test_mail/tests/test_mail_mail.py

```diff
@@ -824,6 +831,97 @@ def test_mail_mail_values_unicode(self):
         self.assertEqual(self._mails[0]['email_to'], ['test.😊@example.com'])
 
 
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_mail_mail_send_server_attachment_to_download_link(self):
+        """ Test that when the mail size exceeds the max email size limit,
+        attachments are turned into download links added at the end of the
+        email content.
+
+        The feature is tested in the following conditions:
+        - using a specified server or the default one (to test command ICP parameter)
+        - in batch mode
+        - with mail that exceed (with one or more attachments) or not the limit
+        - with attachment owned by a business record or not (attachments not owned by a
+        business record are never turned into links because they are deleted after the
+        mail is sent)
+        """
+        def count_attachments(message):
+            return (0 if isinstance(message, str)
+                    else sum(count_attachments(part) for part in message.get_payload()) if message.is_multipart()
+                    else 'attachment' in message.get('Content-Disposition', ''))
+
+        attachment_size_in_bytes = 1024 * 128
+        datas = base64.b64encode((''.join('.' for _ in range(0, attachment_size_in_bytes))).encode())
+        # Define some constant to ease the understanding of the test
+        using_specific_mail_server = self.mail_server_domain_2
+        using_default_mail_server = None
+        cst_max_size_always_exceed = 0.1
+        cst_max_size_never_exceed = 10
+
+        # cst_max... are used to define max_email_size on the server that is not used with value that would
+        # lead to an incorrect expected_is_links if they were used. We do this to check that the right max
+        # email size is used.
+        for (n_attachment, mail_server, default_email_size, max_email_size, business_attachment), expected_is_links in (
+                # 1 attachment which doesn't exceed max size
+                ((1, using_default_mail_server, 0.3, cst_max_size_always_exceed, True), False),
```

Comment:
Would avoid using a variable, better be explicit it is void, like
```suggestion
                ((1, self.env['ir.mail_server'], 0.3, cst_max_size_always_exceed, True), False),
```